kratos/geometries/nurbs_curve_geometry.h

```diff
@@ -0,0 +1,488 @@
+        geometry's points.*/
+    typedef  typename BaseType::PointsArrayType PointsArrayType;
+
+    typedef  typename BaseType::CoordinatesArrayType CoordinatesArrayType;
+
+    /// Counted pointer of NurbsCurveGeometry
+    KRATOS_CLASS_POINTER_DEFINITION(NurbsCurveGeometry);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Conctructor for B-Spline curves
+    NurbsCurveGeometry(
+        const PointsArrayType& rThisPoints,
+        const SizeType PolynomialDegree,
+        const Vector& rKnots)
+        : BaseType(rThisPoints, &msGeometryData)
+        , mPolynomialDegree(PolynomialDegree)
+        , mKnots(rKnots)
+    {
+        KRATOS_DEBUG_ERROR_IF(rKnots.size() != NurbsUtilities::GetNumberOfKnots(PolynomialDegree, rThisPoints.size()))
+            << "Number of knots and control points do not match!" << std::endl;
+    }
+
+    /// Conctructor for NURBS curves
+    NurbsCurveGeometry(
+        const PointsArrayType& rThisPoints,
+        const SizeType PolynomialDegree,
+        const Vector& rKnots,
+        const Vector& rWeights)
+        : BaseType(rThisPoints, &msGeometryData)
+        , mPolynomialDegree(PolynomialDegree)
+        , mKnots(rKnots)
+        , mWeights(rWeights)
+    {
+        KRATOS_DEBUG_ERROR_IF(rKnots.size() != NurbsUtilities::GetNumberOfKnots(PolynomialDegree, rThisPoints.size()))
+            << "Number of knots and control points do not match!" << std::endl;
+
+        KRATOS_DEBUG_ERROR_IF(rWeights.size() != rThisPoints.size())
+            << "Number of control points and weights do not match!" << std::endl;
+    }
+
+    explicit NurbsCurveGeometry(const PointsArrayType& ThisPoints)
+        : BaseType(ThisPoints, &msGeometryData)
+    {
+    }
+
+    /// Copy constructor.
+    NurbsCurveGeometry(NurbsCurveGeometry<TWorkingSpaceDimension, TContainerPointType>  const& rOther)
+        : BaseType(rOther)
+        , mPolynomialDegree(rOther.mPolynomialDegree)
+        , mKnots(rOther.mKnots)
+        , mWeights(rOther.mWeights)
+    {
+    }
+
+    /// Copy constructor from a geometry with different point type.
+    template<class TOtherContainerPointType> NurbsCurveGeometry(
+        NurbsCurveGeometry<TWorkingSpaceDimension, TOtherContainerPointType> const& rOther)
+        : BaseType(rOther, &msGeometryData)
+        , mPolynomialDegree(rOther.mPolynomialDegree)
+        , mKnots(rOther.mKnots)
+        , mWeights(rOther.mWeights)
+    {
+    }
+
+    /// Destructor
+    ~NurbsCurveGeometry() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignment operator.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    NurbsCurveGeometry& operator=(const NurbsCurveGeometry& rOther)
```

Comment:
do you have tests for copy&assignment operator?

Reply:
no we don't